query-planner-js/src/FieldSet.ts

```diff
@@ -87,7 +56,7 @@ export function printFields(fields?: FieldSet) {
  */
 export function debugPrintFields(fields?: FieldSet) : string {
   if (!fields) return '[]';
-  return '[' + fields.map(debugPrintField).join(', ') + ']'
+  return '[' + fields.map(f => debugPrintField(f)).join(', ') + ']'
```

Comment:
Then story is that when I was debugging something I added a 2nd optional boolean argument to `debugPrintField` to selectively show the selection set in the representation, but when I did so, the code above stopped type-checking because the argument of `map` can actually take up to 3 arguments, and the 2nd argument is a `number`, not a `boolean`. So I switched back to the code above. I later remove the flag on `debugPrintField` (the printing ended up ugly and not that useful so figured I wouldn't include it) and just didn't revert this.

With that said, this experience led me to wonder if using arrow functions in this case wasn't a better idea in general. For instance, say I had added a 2nd optional argument to `debugPrintField` that was a `number` (but was not meant to be the index that `map` passes as its callback as 2nd argument). The code would still have type-checked, yet it would not have worked as expected, and that would have been surprising/hard to debug because I just added a new optional argument and hadn't used it yet.

Anyway, I'm considering favouring the use of arrow functions in those cases (and only passing functions directly are arguments when the passed functions has at least as many arguments than expected by the function argument). Opinion?

Reply:
Ah yes, I understand the reasoning now. The additional arguments to the `map` closure can definitely be confusing. I'm not against following this as a rule, but if we do we should probably consider writing it down somewhere or even enforcing it with a linting rule. Could also be a good discussion piece for a company wide style guide.